src/tests/unit_tests/test_vectorstores.py

```diff
@@ -0,0 +1,186 @@
+    docs = firestore_store.collection.stream()
+    for doc in docs:
+        data = doc.to_dict()
+        if doc.id == "1":
+            assert data["content"] == "test_doc1"
+            assert data["metadata"] == {"foo": "bar"}
+        elif doc.id == "2":
+            assert data["content"] == "test_doc2"
+            assert data["metadata"] == {"baz": "qux"}
+
+
+def test_firestore_add_vectors_auto_id():
+    """
+    An end-to-end test for adding vectors to FirestoreVectorStore.
+    """
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Add vectors to Firestore
+    ids = firestore_store.add_texts(
+        ["test_doc_1", "test_doc_2"],
+    )
+
+    # Verify that the first document was added to Firestore
+    # with a generated ID that matches the returned ID
+    doc = firestore_store.collection.document(ids[0]).get()
+    data = doc.to_dict()
+    assert doc.id == ids[0]
+    assert data["content"] == "test_doc_1"
+
+    # Verify that the second document was added to Firestore
+    # with a generated ID that matches the returned ID
+    doc = firestore_store.collection.document(ids[1]).get()
+    data = doc.to_dict()
+    assert doc.id == ids[1]
+    assert data["content"] == "test_doc_2"
+
+
+def test_firestore_add_vectors_assertions():
+    """
+    Tests assertions in FirestoreVectorStore.add_vectors method.
+    """
+
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Test assertions
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts(["test_doc1", "test_doc2"], ids=["1", "2", "3"])
+
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts(
+            ["test_doc1", "test_doc2"], metadatas=[{"foo": "bar"}]
+        )
+
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts(["test_doc1", "test_doc2"], ids=["1"])
+
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts([])
+
+
+def test_firestore_update_vectors():
+    """
+    An end-to-end test for updating vectors in FirestoreVectorStore.
+    """
+
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Add vectors to Firestore
+    firestore_store.add_texts(["test1", "test2"], ids=["1", "2"])
+
+    # Verify that the vectors were updated in Firestore
+    docs = firestore_store.collection.stream()
+    for doc in docs:
+        data = doc.to_dict()
+        if doc.id == "1":
+            assert data["content"] == "test1"
+            assert data["metadata"] == {"foo": "bar"}
+        elif doc.id == "2":
+            assert data["content"] == "test2"
+            assert data["metadata"] == {"baz": "qux"}
+
+
+def test_firestore_similarity_search():
```

Comment:
Can we add a tests were it's visible we are not fetching all documents? 

With this test I only see we are fetching all documents. Although this is expected , I'd like to be certain by adding documents it won't retrieve in the search (?) unless this is not possible

Reply:
Since we're using fake embeddings (which are generated randomly) we cannot assert the values of the search results, as it's inconsistent and random, only the lenght can be verified.
```
class FakeEmbeddings(Embeddings, BaseModel):
    """Fake embedding model."""

    size: int
    """The size of the embedding vector."""

    def _get_embedding(self) -> List[float]:
        return list(np.random.normal(size=self.size))
```